src/Lykke.HftApi.Services/SiriusWalletsService.cs

```diff
@@ -177,32 +165,121 @@ public async Task<DepositWallet> GetWalletAddressAsync(string clientId, string w
+                if (assetById != null)
+                {
+                    result.Add(new DepositWallet
+                    {
+                        AssetId = assetById.AssetId,
+                        Symbol = assetById.Symbol,
+                        State = DepositWalletState.NotFound
+                    });
+                }
+
+                return result;
             }
 
             var account = accountSearchResponse.Body.Items.Single();
 
-            if (account.State == AccountStateModel.Creating)
-            {
-                return new DepositWallet {State = DepositWalletState.Creating};
-            }
-            else if(account.State == AccountStateModel.Blocked)
+            switch (account.State)
             {
-                return new DepositWallet {State = DepositWalletState.Blocked};
-            } else if (account.State == AccountStateModel.Active)
-            {
-                var accountDetailsResponse = await _siriusApiClient.Accounts.GetDetailsAsync(new AccountGetDetailsRequest
+                case AccountStateModel.Creating:
+                    if (assetById != null)
+                        result.Add(new DepositWallet {AssetId = assetById.AssetId, Symbol = assetById.Symbol, State = DepositWalletState.Creating});
+                    break;
+                case AccountStateModel.Blocked:
+                    if (assetById != null)
+                        result.Add(new DepositWallet {AssetId = assetById.AssetId, Symbol = assetById.Symbol, State = DepositWalletState.Blocked});
+                    break;
+                case AccountStateModel.Active:
                 {
-                    AccountId = account.Id,
-                    AssetId = siriusAssetId
-                });
+                    if (siriusAssetId.HasValue)
+                    {
+                        var accountDetailsResponse = await _siriusApiClient.Accounts.GetDetailsAsync(new AccountGetDetailsRequest
+                        {
+                            AccountId = account.Id,
+                            AssetId = siriusAssetId.Value
+                        });
+
+                        if (accountDetailsResponse.ResultCase == AccountGetDetailsResponse.ResultOneofCase.Error)
+                        {
+                            var message = "Error fetching Sirius Account details";
+                            _log.Warning(nameof(GetWalletAddressesAsync),
+                                message,
+                                context: new
+                                {
+                                    error = accountSearchResponse.Error,
+                                    walletId,
+                                    clientId
+                                });
+                            throw new Exception(message);
+                        }
+
+                        var asset = assets.FirstOrDefault(a => a.SiriusBlockchainId == accountDetailsResponse.Body.AccountDetail.BlockchainId);
```

Comment:
It looks like `assetById` can be used here instead of searching the asset one more time by `SiriusBlockchainId`